pkg/daytona/unmarshal.go

```diff
@@ -0,0 +1,251 @@
+		switch f.Kind() {
+		case reflect.Struct:
+			err := UnmarshalSecrets(client, f.Addr().Interface(), WithApex(config.apex))
+			if err != nil {
+				return err
+			}
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			if path == "" {
+				continue
+			}
+
+			var iv int64
+
+			v, err := fetchValue(client, path, valueIndex)
+			if err != nil {
+				return err
+			}
+
+			if f.Kind() == reflect.Int64 && f.Type().String() == "time.Duration" {
+				var d time.Duration
+				dur, ok := v.(string)
+				if ok {
+					d, err = time.ParseDuration(dur)
+					if err != nil {
+						return err
+					}
+					iv = int64(d)
+				} else {
+					return fmt.Errorf("expected a string but was given type %T for field %s", v, fName)
+				}
+			} else {
+				switch v := v.(type) {
+				case json.Number:
+					if value, err := v.Int64(); err == nil {
+						iv = value
+					} else {
+						return err
+					}
+				case string:
+					vv, err := strconv.ParseInt(v, 0, f.Type().Bits())
+					if err != nil {
+						return err
+					}
+					iv = vv
+				default:
+					return fmt.Errorf("expected a number or string but was given type %T for field %s", v, fName)
+				}
+			}
+			f.SetInt(iv)
+		case reflect.Float32, reflect.Float64:
+			if path == "" {
+				continue
+			}
+
+			var iv float64
+
+			v, err := fetchValue(client, path, valueIndex)
+			if err != nil {
+				return err
+			}
+			switch v := v.(type) {
+			case json.Number:
+				if value, err := v.Float64(); err == nil {
+					iv = value
+				} else {
+					return err
+				}
+			case string:
+				vv, err := strconv.ParseFloat(v, f.Type().Bits())
+				if err != nil {
+					return err
+				}
+				iv = vv
+			default:
+				return fmt.Errorf("expected a float or string but was given type %T for field %s", v, fName)
+			}
+			f.SetFloat(iv)
+		case reflect.String:
+			if path == "" {
+				continue
+			}
+			v, err := fetchValue(client, path, valueIndex)
+			if err != nil {
+				return err
+			}
+			if vv, ok := v.(string); ok {
+				f.SetString(vv)
+			} else {
+				return fmt.Errorf("expected a string but was given type %T for field %s", v, fName)
+			}
+		case reflect.Bool:
+			if path == "" {
+				continue
+			}
+			v, err := fetchValue(client, path, valueIndex)
+			if err != nil {
+				return err
+			}
+
+			var b bool
+			switch v := v.(type) {
+			case bool:
+				b = v
+			case string:
+				pb, err := strconv.ParseBool(v)
+				if err != nil {
+					return err
+				}
+				b = pb
+
+			default:
+				return fmt.Errorf("expected a bool or string but was given type %T for field %s", v, fName)
+			}
+			f.SetBool(b)
+		default:
+			continue
+		}
+	}
+	return nil
+}
+
+func parsePath(apex string, tag reflect.StructTag) (path, valueIndex string) {
+	p, ok := tag.Lookup(vaultTagPath)
+	if ok {
+		path = p
+	} else {
+		// try to use an apex
+		if apex != "" {
+			p, ok := tag.Lookup(vaultTagPathKey)
+			if ok {
+				path = fmt.Sprintf("%s/%s", strings.TrimSuffix(apex, "/"), p)
```

Comment:
should we do the ` strings.TrimSuffix(apex, "/")`  as part of the Option setting?  that way its only ran once

Reply:
Agree!